### PR TITLE
Tidy up range of supported IntelliJ versions

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -76,12 +76,17 @@ allprojects {
...
     betterStringsVerson = '0.3'
     checkstyleToolVersion = '8.30'
-    // The following places must also be updated when updating the version below:
-    // * `patchPluginXml.untilBuild` in root Gradle project config in this file
-    // * `runPluginVerifier.ides` in root Gradle project config in this file
-    // * range of UI-tested versions in .circle/config.yml
-    intellijVersion = '2020.1.2'
-    intellijVersionForRunIde = project.properties.intellijVersionForRunIde ?: intellijVersion
+
+    // Update the range of UI-tested versions in .circle/config.yml when any of the below is updated
```
Comment (PawelLipski) [inline build.gradle:80]: Unrel, just a fix for a rare race condition in UI tests

File: testCommon/src/test/java/com/virtuslab/gitmachete/testcommon/BaseGitRepositoryBackedTestSuite.java [hunk 2]
```diff
@@ -63,14 +60,13 @@ private void prepareRepoFromScript(String scriptName) {
     Assert.assertEquals(0, process.exitValue());
   }
 
-  @Rule(order = Integer.MIN_VALUE)
```
Comment (PawelLipski) [inline testCommon/src/test/java/com/virtuslab/gitmachete/testcommon/BaseGitRepositoryBackedTestSuite.java:66]: Unrel, just a fix for a rare race condition in UI tests

File: build.gradle (hunk 1 above)
Comment (PawelLipski) [inline build.gradle:80]: Now the versions are in just 2 places, not 4 :trollface:

File: build.gradle [hunk 3]
```diff
@@ -439,8 +441,12 @@ apply from: 'https://raw.githubusercontent.com/FWDekker/intellij-randomness/mast
 
 runPluginVerifier {
   pluginFileName = "${intellij.pluginName}-${version}"
-  ides = [ 'IC-2019.3', 'IC-2020.1', 'IC-2020.1.2' ]
   verifierVersion = '1.241'
+  ides = [
+    intellijVersions.earliestSupported,
+    *intellijVersions.inBetweenForCompatibilityChecks,
```
Comment (micpiotrowski) [inline build.gradle:447]: What does this asterisk mean?

File: scripts/run-ui-tests [hunk 4]
```diff
@@ -8,39 +8,52 @@ source "$self_dir"/utils.sh
...
 # but we still need to kill the IDE instance in case that test was not successful AND the IDE is still running.
-function finish {
-  if ps -p $ide_pid >/dev/null 2>&1; then
+function finish() {
+  if is_ide_running; then
     kill $ide_pid
   fi
+  # Let's wait until IDE process is terminated to avoid any risk of race conditions
+  # when another UI test is to be run right after this script completes.
+  while is_ide_running; do
```
Comment (micpiotrowski) [inline scripts/run-ui-tests:35]: Maybe some counter to avoid infinity loop in some strange cases?

Comment (PawelLipski) [reply scripts/run-ui-tests:35]: Hmmm :thinking: TBH I think ideally we could use the "graceful shutdown" pattern: SIGINT, 10 seconds or so for the process to shut down, then SIGKILL if it's "rogue" (still running after 10 seconds)

Comment (PawelLipski) [reply scripts/run-ui-tests:35]: Will do on this or another PR

File: build.gradle (hunk 3 above)
Comment (PawelLipski) [reply build.gradle:447]: Expand the list

Comment (PawelLipski) [reply build.gradle:447]: So `[a, *b, c]` is pretty much `[a] + b + [c]`

**Pull request merged**